asv_bench/benchmarks/period.py

```diff
@@ -0,0 +1,9 @@
+from pandas import Period, PeriodIndex, date_range
+
+
+class create_period_index_from_date_range(object):
+    goal_time = 0.2
+
+    def time_period_index(self):
+        # Simulate irregular PeriodIndex
```

Comment:
can you add a benchmark for #11831 as well (you can put in this file even though its plotting)